### ISSUE TITLE
improve efficiency of loop_tasks is_current check

### ISSUE THREAD

**Issue by aappling-usgs** [DOI-USGS/scipiper#65]
I was just looking at https://github.com/USGS-R/scipiper/blob/master/R/loop_tasks.R#L70-L76 again and am not sure it's as efficient as I initially thought. In particular, we call `get_remake_status(targets, task_makefile)` on _all_ possible targets and only later exclude those that already have an existing file. But the check for whether existing files are current is the slow part, so we surely shouldn't be calling `get_remake_status(targets, task_makefile)` on those existing files at all, right?

And maybe we should make it a user option to either loop over all tasks or to loop over complete unbuilt tasks (for which file doesn't yet exist)? Sometimes the hashing is expensive while the retries of unbuilt files are the focus; other times the hashing is cheap and we'd like to include retries for built but out-of-date files (e.g., when we forget or don't know to or don't want to delete the files before building).

**Comment by jread-usgs** [comment]
Yes, I think I ran into this taking a long time to run when I had a lot of targets. I wasn't thinking much about the inefficiencies in `which_incomplete` as much as us needing to call that again with each `try` in the loop (I am dealing w/ a failure-prone build). Since we are catching errors within the try, we know which targets succeed and which fail, so can `incomplete_targets <- which_incomplete(targets, task_makefile)` be set once before the first loop and then updated within the `try`s after each `scmake` call?

**Comment by aappling-usgs** [comment]
Ooh, that's also a good idea!

**Comment by jread-usgs** [comment]
I've got this (call `which_incomplete` only once) in the work I am doing for `progress` bar if you want to take a look at it w/ that PR. 

I still think there are gains to be made in the internals of `which_incomplete`
